File: programmers/level1/문자열_내_마음대로_정렬하기.java
package programmers;
import java.util.Arrays;
import java.util.Comparator;
public class 문자열_내_마음대로_정렬하기 {
	static class Solution {
		  public String[] solution(String[] strings, int n) {
		      Arrays.sort(strings, new Comparator<String>(){
				@Override
				public int compare(String o1, String o2) {
					if(o1.charAt(n) < o2.charAt(n)) {
						return -1;
					}else if(o1.charAt(n) > o2.charAt(n)) {
						return 1;
					}
					return o1.compareTo(o2);
				}
		      });
		      return strings;
		  }
		}
	public static void main(String[] args) {
		String[] strings = {"sun", "bed", "car"};
		int n = 1;
		Solution sol = new Solution();
		System.out.println(Arrays.toString(sol.solution(strings, n)));
	}
}

File: programmers/js/new/level1/문자열_내_마음대로_정렬하기.js
function solution(strings, n) {
    for (let i = 0; i < strings.length; ++i) {
        strings[i] = strings[i].charAt(n) + strings[i];
    }
    strings.sort();
    for (let i = 0; i < strings.length; ++i) {
        strings[i] = strings[i].substring(1);
    }
    return strings;
}
